source/Engine.cpp

```diff
@@ -1569,8 +1603,14 @@ void Engine::CalculateStep()
 	FillCollisionSets();
 
 	// Perform collision detection.
-	for(Projectile &projectile : projectiles)
-		DoCollisions(projectile);
+	// We store the visuals in a separate list for every thread to reduce lock contention.
+	vector<pair<mutex, list<Visual>>> parallelBuffer(thread::hardware_concurrency());
+	for_each(execution::par, projectiles.begin(), projectiles.end(), [&](auto &projectile) {
+		DoCollisions(parallelBuffer, projectile);
+	});
+	for(auto &item : parallelBuffer)
```

Comment:
```suggestion

	for(const auto &item : parallelBuffer)
```
they do not get changed, right?